### PR TITLE
feat(components): [select] & [select-v2] add fallbackPlacements prop

### LINKED ISSUE #15605: [Component] [select] el-select 在靠近屏幕边框的时候会出现弹出框方向不对的情况
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.10`
- Element Plus Version: `2.4.3`
- Browser / OS: `e.g. Chrome 96.0.4664.45 / Windows `
- Build Tool: `Vite`

### Related Component

- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XHJcbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xyXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcclxuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcclxuXHJcbmNvbnN0IG1zZyA9IHJlZignSGVsbG8gV29ybGQhJyk7XHJcbmNvbnN0IHNlbGVjdCA9IHJlZigxKTtcclxuPC9zY3JpcHQ+XHJcblxyXG48dGVtcGxhdGU+XHJcbiAgPGgxPnt7IG1zZyB9fTwvaDE+XHJcbiAgPGRpdiBzdHlsZT1cIm1hcmdpbjotNXB4IDtcIj5cclxuICAgIDxlbC1yb3c+XHJcbiAgICAgIDxlbC1jb2wgOnNwYW49XCIzXCIgc3R5bGU9XCJoZWlnaHQ6IDEwMCVcIj48L2VsLWNvbD5cclxuICAgICAgPGVsLWNvbCA6c3Bhbj1cIjE3XCIgc3R5bGU9XCJoZWlnaHQ6IDEwMCVcIj48L2VsLWNvbD5cclxuICAgICAgPGVsLWNvbCA6c3Bhbj1cIjRcIj5cclxuICAgICAgICA8ZWwtc2VsZWN0IHYtbW9kZWw9XCJzZWxlY3RcIiA6c3R5bGU9XCJ7IHdpZHRoOiAnMTAwJScgfVwiIHBsYWNlbWVudD1cImJvdHRvbVwiIHBsYWNlaG9sZGVyPVwi6K+36L6T5YWl5oiW6YCJ5oup5a2X5YW45YC8XCI+XHJcbiAgICAgICAgICA8ZWwtb3B0aW9uIGxhYmVsPVwiMVwiIHZhbHVlPVwiMlwiIC8+XHJcbiAgICAgICAgPC9lbC1zZWxlY3Q+XHJcbiAgICAgIDwvZWwtY29sPlxyXG4gICAgPC9lbC1yb3c+XHJcbiAgPC9kaXY+XHJcbiAgPHA+XHJcbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxyXG4gICAgRWxlbWVudCBQbHVzIHt7IGVwVmVyc2lvbiB9fSArIFZ1ZSB7eyB2dWVWZXJzaW9uIH19XHJcbiAgPC9wPlxyXG48L3RlbXBsYXRlPlxyXG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

见连接

### What is Expected?

能够正常弹框

### What is actually happening?

弹框位置与设置位置不一致

### Additional comments

_(empty)_

image

### PR DESCRIPTION
**Pull request by Fuphoenixes**
Please make sure these boxes are checked before submitting your PR, thank you!
fixed: https://github.com/element-plus/element-plus/issues/15605
- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/en-US/component/select-v2.md [hunk 1]
```diff
@@ -201,53 +201,54 @@ select-v2/custom-loading
...
+| ----------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------- |
+| model-value / v-model               | biding value                                                                                                                             | ^[string] / ^[number] / ^[boolean] / ^[object] / ^[array]                                                                                                                   | —                                              |
```
Comment (kooriookami) [inline docs/en-US/component/select-v2.md:206]:
```suggestion
| model-value / v-model               | binding value                                                                                                                             | ^[string] / ^[number] / ^[boolean] / ^[object] / ^[array]                                                                                                                   | —                                              |
```

File: docs/en-US/component/select-v2.md [hunk 2]
```diff
@@ -201,53 +201,54 @@ select-v2/custom-loading
...
+| placement                           | position of dropdown                                                                                                                     | ^[enum]`'top' \| 'top-start' \| 'top-end' \| 'bottom' \| 'bottom-start' \| 'bottom-end' \| 'left' \| 'left-start' \| 'left-end' \| 'right' \| 'right-start' \| 'right-end'` | bottom-start                                   |
+| fallback-placements ^(2.5.5)        | list of possible positions for dropdown [popper.js](https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements)                    | ^[arrary]`Placement[]`                                                                                                                                                      | ['bottom-start', 'top-start', 'right', 'left'] |
```
Comment (kooriookami) [inline docs/en-US/component/select-v2.md:242]:
```suggestion
| fallback-placements ^(2.5.6)        | list of possible positions for dropdown [popper.js](https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements)                    | ^[arrary]`Placement[]`                                                                                                                                                      | ['bottom-start', 'top-start', 'right', 'left'] |
```

File: docs/en-US/component/select.md [hunk 3]
```diff
@@ -163,47 +163,48 @@ select/custom-loading
...
+| validate-event                  | whether to trigger form validation                                                                                    | ^[boolean]                                                                                                                                                                  | true                                           |
+| placement ^(2.2.17)             | position of dropdown                                                                                                  | ^[enum]`'top' \| 'top-start' \| 'top-end' \| 'bottom' \| 'bottom-start' \| 'bottom-end' \| 'left' \| 'left-start' \| 'left-end' \| 'right' \| 'right-start' \| 'right-end'` | bottom-start                                   |
+| fallback-placements ^(2.5.5)    | list of possible positions for dropdown [popper.js](https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements) | ^[arrary]`Placement[]`                                                                                                                                                      | ['bottom-start', 'top-start', 'right', 'left'] |
```
Comment (kooriookami) [inline docs/en-US/component/select.md:204]:
```suggestion
| fallback-placements ^(2.5.6)    | list of possible positions for dropdown [popper.js](https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements) | ^[arrary]`Placement[]`                                                                                                                                                      | ['bottom-start', 'top-start', 'right', 'left'] |
```

File: docs/en-US/component/tooltip.md [hunk 4]
```diff
@@ -157,6 +157,7 @@ tooltip/animations
...
 | raw-content               | whether `content` is treated as HTML string                                                                                                             | ^[boolean]                                                                                                                                                                  | false             |
 | placement                 | position of Tooltip                                                                                                                                     | ^[enum]`'top' \| 'top-start' \| 'top-end' \| 'bottom' \| 'bottom-start' \| 'bottom-end' \| 'left' \| 'left-start' \| 'left-end' \| 'right' \| 'right-start' \| 'right-end'` | bottom            |
+| fallback-placements       | list of possible positions for Tooltip [popper.js](https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements)                                    | ^[arrary]`Placement[]`                                                                                                                                                      | —                 |
```
Comment (kooriookami) [inline docs/en-US/component/tooltip.md:160]: Does this need version tag?

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: docs/en-US/component/tooltip.md (hunk 4 above)
Comment (Fuphoenixes) [reply docs/en-US/component/tooltip.md:160]: It shouldn't be necessary. It already supports it, but the documentation is not written.

Comment (kooriookami) [reply docs/en-US/component/tooltip.md:160]: Alright

**Review by btea — APPROVED** [review]

**Review by kooriookami — APPROVED** [review]

**Comment by kooriookami** [comment]
Can you add this prop to `cascader` too?

**Pull request merged**